pkg/dedup/iter.go

```diff
@@ -160,9 +167,13 @@ func (s *dedupSeriesSet) At() storage.Series {
 	// Clients may store the series, so we must make a copy of the slice before advancing.
 	repl := make([]storage.Series, len(s.replicas))
 	copy(repl, s.replicas)
-	if s.f == UseMergedSeries {
+	if s.deduplicationFunc == AlgorithmQuorum {
 		// merge all samples which are ingested via receiver, no skips.
-		return NewMergedSeries(s.lset, repl)
+		// feed the merged series into dedup series which apply counter adjustment
+		return NewMergedSeries(s.lset, repl, s.f)
+	}
+	if s.deduplicationFunc == AlgorithmChain {
```

Comment:
fyi @yuchen-db, I've also added the prometheus implementation, but it doesn't work for a number of unit tests if you wonder